include/circt/Dialect/Verif/VerifOps.td

```diff
@@ -169,6 +169,46 @@ def HasBeenResetOp : VerifOp<"has_been_reset", [Pure]> {
 // Logical Equivalence Checking and Model Checking
 //===----------------------------------------------------------------------===//
 
+def BMCOp : VerifOp<"bmc", [
+  IsolatedFromAbove,
+  SingleBlockImplicitTerminator<"verif::YieldOp">,
+]> {
+  let summary = "Perform a bounded model check";
+  let description = [{
+    This operation represents a bounded model checking problem explicitly in
+    the IR. The `circuit` region contains the circuit (alongside the `verif`
+    property checking operations) to be checked.
+
+    The `init` region contains the logic to initialize the clock signals, and
+    will be executed once before any other region - it cannot take any
+    arguments, and should return as many `!seq.clock` values as the `circuit`
+    region has `!seq.clock` arguments, followed by any initial arguments of
+    'state' arguments to be fed to the `loop` region (see below).
+
+    The `loop` region contains the logic to advance the clock signals, and will
+    be executed after each execution of the `circuit` region. It must have all
+    the arguments that the `circuit` region has, but these can be followed by
+    additional 'state' arguments to represent e.g. which clock should be
+    toggled next. The `loop` region should yield as many `!seq.clock` values as
+    the `circuit` region has `!seq.clock` arguments, followed by the updated
+    values of all 'state' arguments.
```

Comment:
Agreed this would be useful, and I was thinking about it a lot while I designed the op, but ultimately decided it would probably make sense for a later PR. The main tricky bit there is the signatures of the init and loop regions - it would be nice to avoid them operating on SMT types in my opinion, since it's not been lowered to SMT yet and we then maintain a clear correspondence with values in the circuit region, which makes creating/manipulating symbolic values complicated. The two workarounds I can think of are:
1. Once the verif intent ops make it upstream, we hand full control of input manipulation over to the op (i.e. we don't implicitly create a new SMT bitvector for each variable on each cycle). We could then allow some `verif.symbolic_val` op (here, I mean the version that maps to an SMT variable, not a BTOR2 input) to define symbolic values that are then given to the circuit region. This is my preferred option, because it may also mean the loop region could constrain the value so you have very powerful control over the executions traversed if you want it.
2. Slightly hacky, but we could also always generate a new SMT value for each variable, then give them all to the loop region (keeping the signature it has now) on each cycle. The loop region can then either return it untouched (keeping it symbolic) or return a concrete value in its place. This avoids any type trickiness since the symbolic variables aren't actually created or manipulated in the loop region, but it could lead to a lot of unnecessary calls to the solver to create new BVs, and doesn't have the control advantage I mention above.